File: src/main/java/com/javakart/service/ProductService.java
package com.javakart.service;

import com.javakart.dto.ProductDTO;
import java.util.List;

public interface ProductService {
    ProductDTO createProduct(ProductDTO productDTO);
    ProductDTO updateProduct(Long productId, ProductDTO productDTO);
    void deleteProduct(Long productId);
    ProductDTO getProductById(Long productId);
    List<ProductDTO> getAllProducts();
    List<ProductDTO> getProductsByCategory(Long categoryId);
    List<ProductDTO> searchProducts(String keyword);
}

File: javakart-frontend/src/services/productService.js
// src/services/productService.js
import { productAPI } from './api';

export const productService = {
  getAllProducts: async () => {
    try {
      const response = await productAPI.getAll();
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  getProductById: async (id) => {
    try {
      const response = await productAPI.getById(id);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  searchProducts: async (keyword) => {
    try {
      const response = await productAPI.search(keyword);
      return response.data;
    } catch (error) {
      throw error;
    }
  },

  getProductsByCategory: async (categoryId) => {
    try {
      const response = await productAPI.getByCategory(categoryId);
      return response.data;
    } catch (error) {
      throw error;
    }
  }
};